powerapi-cli/src/main/scala/org/powerapi/app/Application.scala

```diff
@@ -0,0 +1,222 @@
+  val durationR = """\d+""".r
+  val pidR = """(\d+)""".r
+  val appR = """(.+)""".r
+
+  @volatile var powerMeters = Seq[PowerMeter]()
+  @volatile var monitors = Seq[PowerMonitoring]()
+
+  val shutdownHookThread = scala.sys.ShutdownHookThread {
+    println("It's the time for sleeping! ...")
+
+    monitors.foreach(monitor => monitor.cancel())
+    monitors = Seq()
+    powerMeters.foreach(powerMeter => powerMeter.shutdown())
+    powerMeters = Seq()
+  }
+
+  def validateModules(str: String) = str match {
+    case modulesR(_*) => true
+    case _ => false
+  }
+
+  implicit def modulesStrToPowerModules(str: String): Seq[PowerModule] = {
+    (for(module <- str.split(",")) yield {
+      module match {
+        case "cpu-simple" => CpuSimpleModule()
+        case "cpu-dvfs" => CpuDvfsModule()
+        case "libpfm-core" => LibpfmCoreModule()
+        case "libpfm-core-process" => LibpfmCoreProcessModule()
+        case "powerspy" => PowerSpyModule()
+      }
+    }).toSeq
+  }
+
+  def validateAgg(str: String): Boolean = str match {
+    case aggR(_*) => true
+    case _ => false
+  }
+
+  implicit def aggStrToAggFunction(str: String): Seq[Power] => Power = {
+    str match {
+      case "max" => MAX
+      case "min" => MIN
+      case "geomean" => GEOMEAN
+      case "logsum" => LOGSUM
+      case "mean" => MEAN
+      case "median" => MEDIAN
+      case "stdev" => STDEV
+      case "sum" => SUM
+      case "variance" => VARIANCE
+    }
+  }
+
+  def validateDuration(str: String): Boolean = str match {
+    case durationR(_*) => true
+    case _ => false
+  }
+
+  implicit def targetsStrToTargets(str: String): Seq[Target] = {
+    val strTargets = if(str.split(",").contains("all")) {
+      "all"
+    }
+    else str
+
+    (for(target <- strTargets.split(",")) yield {
+      target match {
+        case "" => Process(ManagementFactory.getRuntimeMXBean.getName.split("@")(0).toInt)
+        case "all" => All
+        case pidR(pid) => Process(pid.toInt)
+        case appR(app) => Application(app)
+      }
+    }).toSeq
+  }
+
+  def printHelp(): Unit = {
+    val str =
+      """
+        |PowerAPI (c) Spirals Team"
+        |
+        |Build a software-defined power meter. Don't forget to configure correctly the modules (see the documentation).
+        |
+        |usage: ./powerapi-cli modules [cpu-simple|cpu-dvfs|libpfm-core|libpfm-core-proces|powerspy, ...]
+        |                              monitor --frequency [ms] --targets [pid, ..., app, ...)|all] --agg [max|min|geomean|logsum|mean|median|stdev|sum|variance] --[console,file [filepath],chart]
+        |                      duration [s]
+        |
+        |example: ./powerapi-cli modules cpu-simple monitor --frequency 1000 --targets firefox --agg max --console monitor --targets chrome --agg max --console
+        |                        modules powerspy monitor --frequency 1000 --targets all -agg max --console
+        |                        duration 30
```

Comment:
Yes, it is an example of use. Here it creates two PowerMeter. One with cpu-simple, and another one with powerspy. Two monitorings will be attached to the first one, one for the second.
The application will be run during 30 seconds (for all PowerMeter)